### ISSUE TITLE
[Feature Request] Add license to Image Metadata as mandatory attribute

### ISSUE THREAD

**Issue by anjastrunk** [SovereignCloudStack/standards#369]
Image metadata support information about

- `license_included` (boolean) indicates whether or not the flavor fee includes the licenses required to use this image. 
- `license_required` (boolean) indicates whether or not a customer must bring its own license to be license compliant. 

Beside these information, content of image's license would be important for customers. Especially, if image includes a non popular operating system. 
I suggest to add an additional metadata called `license` as mandatory metadata `license` MUST refer to a SPDX identifiers or URL to license document.

**Comment by berendt** [comment]
Makes sense to me. I think it is a good idea to also have the `license` metadata as MUST.

**Comment by garloff** [comment]
If you read the [current standard](https://docs.scs.community/standards/scs-0102-v1-image-metadata#licensing--maintenance-subscription--support) as a user you could conclude that you can use the image without any extra fees if `license_required` is not set. (If you want maintenance, you need to also ensure `subscription_required` is not set.)

It would be nice to explicitly state a license for an image.
However, to the best of my knowledge, no such thing exists.
A Linux distribution is a *collection* of software which may be under a variety of licenses.
The distributor's job is to ensure that
(a) no included software interacts with other included software that is not allowed by the license, e.g. linking proprietary code to GPL code
(b) users have the right to use all included software -- if special strings are attached, display the EULAs in the install process
(c) the whole thing can be distributed

To my knowledge, there is no nice solution for (b) with images, so images should not include software that requires EULAs to be displayed.
The image build needs to take care of (a) and (c).

If we wanted to, we could try to collect a list of licenses of all code included in images and then have a long list in the license metadata. To my knowledge noone is doing this currently. If we go there, we should be aware that this is non-trivial to get right, technically and legally.

**Comment by fkr** [comment]
I will close this as not feasable.
